addons/l10n_ke_edi_tremol/static/src/js/send_invoice.js

```diff
@@ -6,28 +6,43 @@ odoo.define('l10n_ke_edi_tremol.action_post_send_invoice', function (require) {
     var _t = core._t;
 
     async function post_send(parent, {params}) {
-        const move_id = params.move_id;
-        await ajax.post(params.proxy_address + '/hw_proxy/l10n_ke_cu_send', params).then(function (res) {
-            const res_obj = JSON.parse(res);
-            if (res_obj.status != "ok") {
-                Dialog.alert(this, "Posting the invoice has failed, with the message: \n" + res_obj.status);
-            } else {
-                rpc.query({
-                    model: 'account.move',
-                    method: 'l10n_ke_cu_response',
-                    args: [[], {'replies': res_obj.replies, 'serial_number': res_obj.serial_number, 'move_id': move_id}],
-                }).then(function () {
-                    parent.services.action.doAction({
-                        'type': 'ir.actions.client',
-                        'tag': 'reload',
-                    });
-                }, function () {
-                    Dialog.alert(this, _t("Error trying to connect to Odoo. Check your internet connection"));
-                })
+        let refresh = false;
+        for (let move_id in params.invoices) {
+            try {
+                const res = await ajax.post(
+                    params.invoices[move_id].proxy_address + '/hw_proxy/l10n_ke_cu_send', {
+                        messages: params.invoices[move_id].messages,
+                        company_vat: params.invoices[move_id].company_vat
+                    }
+                );
+                const res_obj = JSON.parse(res);
+                if (res_obj.status === "ok") {
+                    try {
+                        await rpc.query({
+                            model: 'account.move',
+                            method: 'l10n_ke_cu_response',
+                            args: [[], {'replies': res_obj.replies, 'serial_number': res_obj.serial_number, 'move_id': move_id}],
+                        });
+                        refresh = true;
+                    } catch (e) {
+                        Dialog.alert(this, _t("Error trying to connect to Odoo. Check your internet connection"));
+                        break;
+                    }
+                } else {
+                    Dialog.alert(this, _t("Posting an invoice has failed, with the message: \n") + res_obj.status);
+                    break;
+                }
+            } catch(e) {
+                Dialog.alert(this, _t("Error trying to connect to the middleware. Is the middleware running?"));
+                break;
             }
-        }, function () {
-            Dialog.alert(this, _t("Error trying to connect to the middleware. Is the middleware running?"));
-        })
+        }
+        if (refresh) {
```

Comment:
Not sure what the goal is here, because I would suppose the for loop would already handle everything

Reply:
I'm not sure what you mean here.
The refresh is there so that the user can see the updated fields

Reply:
Somehow I thought it was still inside of the for-loop somehow...  But did not it do the refresh before on the invoice form or is this only needed for the tree view?

Reply:
This refresh was here before, but it did the refresh upon performing the rpc call to update the `cu` fields. It has to be moved outside of the loop because refreshing interrupts the client action.